portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java

```diff
@@ -257,6 +251,11 @@ public DLFolder deleteFolder(
 
 		deleteFolderDependencies(dlFolder, includeTrashedEntries);
 
+		if (dlFolder.isApproved()) {
```

Comment:
We need to check that `dlFolder.getParentFolderId()` is not 0 because otherwise this will fail?

Reply:
You're right, I will add the other condition to the if statement.